src/transformers/generation/candidate_generator.py

```diff
@@ -157,6 +157,15 @@ def __init__(
         self.generation_config.return_dict_in_generate = True
         self.generation_config.output_scores = True
 
+        # avoid unnecessary warnings that min_length is larger than max_new_tokens
+        input_length = input_ids.shape[-1]
+        min_new_tokens = (
+            self.generation_config.min_new_tokens if self.generation_config.min_new_tokens is not None else 0
+        )
+        self.min_length = max(self.generation_config.min_length, input_length + min_new_tokens)
+        self.generation_config.min_length = 0
+        self.generation_config.min_new_tokens = None
```

Comment:
In light of the most recent changes: before these lines, `self.generation_config.min_length` will hold the main model `min_length` with `min_new_tokens` factored in. As such, we only need to hold the original `min_length`. I'd also recommend holding it under a more descriptive name

```suggestion
        # avoid unnecessary warnings that min_length is larger than max_new_tokens
        self.main_model_min_length = self.generation_config.min_length
        self.generation_config.min_length = 0
        self.generation_config.min_new_tokens = None
```

Reply:
right, I forgot that it all started from speculative decoding 😄